pkg/ovs/openflow/ofctrl_builder_test.go

```diff
@@ -15,30 +15,2171 @@
+			assert.Equal(t, 1, len(flowMod.Match.Fields))
+			checkMatchField(t, tc.expectedMatchField, flowMod.Match.Fields[0])
+			assert.Contains(t, getFlowModMatch(flowMod), tc.expectedMatcherStr)
+		}
+	})
+	t.Run("MatchARPSha/MatchARPTha", func(t *testing.T) {
+		mac1, _ := net.ParseMAC("aa:bb:cc:dd:ee:ff")
+		mac2, _ := net.ParseMAC("aa:bb:cc:dd:ee:fe")
+
+		testCases := []struct {
+			isSrc              bool
+			mac                net.HardwareAddr
+			expectedMatcherStr string
+			expectedMatchField *matchField
+		}{
+			{
+				isSrc:              true,
+				mac:                mac1,
+				expectedMatcherStr: "arp_sha=aa:bb:cc:dd:ee:ff",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					field: openflow15.OXM_FIELD_ARP_SHA,
+					value: &openflow15.ArpXHaField{ArpHa: mac1},
+				},
+			},
+			{
+				isSrc:              true,
+				mac:                mac2,
+				expectedMatcherStr: "arp_sha=aa:bb:cc:dd:ee:fe",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					field: openflow15.OXM_FIELD_ARP_SHA,
+					value: &openflow15.ArpXHaField{ArpHa: mac2},
+				},
+			},
+			{
+				mac:                mac1,
+				expectedMatcherStr: "arp_tha=aa:bb:cc:dd:ee:ff",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					field: openflow15.OXM_FIELD_ARP_THA,
+					value: &openflow15.ArpXHaField{ArpHa: mac1},
+				},
+			},
+			{
+				mac:                mac2,
+				expectedMatcherStr: "arp_tha=aa:bb:cc:dd:ee:fe",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					field: openflow15.OXM_FIELD_ARP_THA,
+					value: &openflow15.ArpXHaField{ArpHa: mac2},
+				},
+			},
+		}
+		for _, tc := range testCases {
+			var fb *ofFlowBuilder
+			if tc.isSrc {
+				fb = table.BuildFlow(1).MatchARPSha(tc.mac).(*ofFlowBuilder)
+				assert.Equal(t, tc.mac, *fb.ofFlow.Match.ArpSha)
+			} else {
+				fb = table.BuildFlow(1).MatchARPTha(tc.mac).(*ofFlowBuilder)
+				assert.Equal(t, tc.mac, *fb.ofFlow.Match.ArpTha)
+			}
+
+			flowMod := getFlowMod(t, fb.Done())
+			assert.Equal(t, 1, len(flowMod.Match.Fields))
+			checkMatchField(t, tc.expectedMatchField, flowMod.Match.Fields[0])
+			assert.Contains(t, getFlowModMatch(flowMod), tc.expectedMatcherStr)
+		}
+	})
+	t.Run("MatchARPSpa/MatchARPTpa", func(t *testing.T) {
+		testCases := []struct {
+			isSrc              bool
+			ip                 net.IP
+			expectedMatcherStr string
+			expectedMatchField *matchField
+		}{
+			{
+				isSrc:              true,
+				ip:                 net.ParseIP("1.1.1.1"),
+				expectedMatcherStr: "arp_spa=1.1.1.1",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					field: openflow15.OXM_FIELD_ARP_SPA,
+					value: &openflow15.ArpXPaField{ArpPa: net.ParseIP("1.1.1.1")},
+				},
+			},
+			{
+				isSrc:              true,
+				ip:                 net.ParseIP("2.2.2.2"),
```

Comment:
Same question here, any specific test case need to be covered by two different IPv4 addresses?

Reply:
Removed one.